consensus/fast-sync/src/fast_sync.rs

```diff
@@ -31,45 +48,78 @@ fn max_height() -> u64 {
     (HASHES_OF_HASHES.len() * BATCH_SIZE) as u64
 }
 
-pub enum FastSyncRequest {
-    ValidateHashes {
-        start_height: u64,
-        block_ids: Vec<BlockId>,
-    },
-}
-
 #[derive(Debug, PartialEq)]
 pub struct ValidBlockId(BlockId);
 
 fn valid_block_ids(block_ids: &[BlockId]) -> Vec<ValidBlockId> {
     block_ids.iter().map(|b| ValidBlockId(*b)).collect()
 }
 
+pub enum FastSyncRequest {
+    ValidateHashes {
+        start_height: u64,
+        block_ids: Vec<BlockId>,
+    },
+    ValidateBlock {
+        block: Box<Block>,
```

Comment:
I think I would rather put `#[allow(clippy::large_enum_variant)]` on this enum than block the block